frontend/src/modules/entitieslist/components/EntitiesList.tsx

```diff
@@ -38,19 +38,26 @@ type InternalProps = Props & {
     store: Record<string, any>;
 };
 
+type State = {
+    entitiesWithActiveSiblings: Array<number>;
+};
+
 /**
  * Displays a list of entities and their current translation.
  *
  * This component will fetch entities when loaded, then display those
  * entities. It interacts with `core/navigation` when an entity is selected.
  *
  */
-export class EntitiesListBase extends React.Component<InternalProps> {
+export class EntitiesListBase extends React.Component<InternalProps, State> {
     list: { current: any };
 
     constructor(props: InternalProps) {
         super(props);
         this.list = React.createRef();
+        this.state = {
+            entitiesWithActiveSiblings: [],
```

Comment:
I wonder why did you decide to keep the state of all entities with revealed siblings in the parent component, instead of in `Entity`?

It should be simpler to keep the logic in the `Entity` component, without the need to passing it downwards from the parent `EntitiesList` component.

Reply:
I couldn't explain it properly in text so I made a video 
https://www.loom.com/share/e6f681e83f2748a9ad9e733b31bc5ed8

Reply:
That's a great explanation, nice work! Do you might pushing or pasting a patch with the code used in the video?

Reply:
My first bet would be that we use index as the `key`, which is problematic when order of items in the list changes (and in our cases it does, because we inject):
https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

Can you try using `entity.pk` instead of `i` as the `key` attribute of the `Entity` component?

Reply:
@Mitch4sho  After thinking on this some more I realized that when you and I worked on this I got too focused on 'finding a working solution' and I should have taken a moment to reflect more on WHY the state was changing based on the index location and thus changing the state for the newly injected sibling instead of the correct entity.

When taking that time, I saw that within the `map()` the `key` is being set to the index number. Since the `key` is what [React](https://reactjs.org/docs/lists-and-keys.html#keys) uses to identify which element to apply the state change to it would then apply the change to the newly injected sibling at that index/key location.

By setting the `key` to the `entity.pk` you should be able to successfully handle the state in `Entity` in the way you demonstrate in your video.

As a side note, this beautifully demonstrates why it is recommended to use a unique identifier other than index whenever possible 😁

Reply:
This makes so much sense now thinking about it! Thank you should be able to complete the refactor tomorrow!